samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -94,6 +95,36 @@ public ZkKeyBuilder getKeyBuilder() {
     return keyBuilder;
   }
 
+  public static class ProcessorData {
```

Comment:
I don't think this static class is necessarily. All you are trying to do here is parse the string provided by the caller to registerProcessorAndGetId. Based on the current code usage, you are using the constructor more as a util method. I don't see a need for a static class here.